src/foam/nanos/dig/DigWebAgent.java

```diff
@@ -339,11 +342,64 @@ public void execute(X x) {
         }
         out.println(returnMessage);
       } else if ( Command.select == command ) {
+        resp.setContentType("text/html");
+
         PropertyInfo idProp = (PropertyInfo) cInfo.getAxiomByName("id");
         ArraySink sink = (ArraySink) ( ! SafetyUtil.isEmpty(id) ?
           dao.where(MLang.EQ(idProp, id)).select(new ArraySink()) :
           dao.select(new ArraySink()));
 
+        SinkParser sinkParser = new SinkParser(cInfo);
+        StringPStream ps = new StringPStream(fields);
+        ParserContextImpl x_ = new ParserContextImpl();
+        ps = (StringPStream) sinkParser.parse(ps, x_);
+
+        Object[] psArray = (Object[]) ps.value();
+        logger.debug("ps.value.length: " + psArray.length);
+
+        for ( Object objPs : psArray ) {
+          logger.debug("ps.value() : " + objPs);
+
+          if ( objPs.toString().startsWith("groupBy") || objPs.toString().startsWith("min") || objPs.toString().startsWith("max") || objPs.toString().startsWith("avg") ) {
```

Comment:
The list of support aggregate operation should be stored in an array and you should match against the whole array. Matching should be done with a case insensitive startsWith instead of just startsWith.